### DISCUSSION TITLE
Add option for "own map design"

### DISCUSSION THREAD

**Discussion by DIrgi** [devemux86/cruiser#560] [Q&A] [answered]
So far I have to add/load a own map design via map/external map design. 

Having an own option point in the selection like "own map design" would be fine.
![image](https://github.com/devemux86/cruiser/assets/57968526/253394a8-a65e-45b9-a87c-b321e5cf6a59)

It could be setup once in options (the path) and then always be available for switching and activating 
under the point "own map design"

**Comment by devemux86** [comment]
@DIrgi thanks for the suggestion.

This seems useful, we can implement something like this.

**Comment by MotoUKRider** [comment, marked as answer]
Do I have the wrong end of the stick, but do you mean something like:

![cruiser-map-theme-load](https://github.com/devemux86/cruiser/assets/125696109/6fb3ce90-30e6-4b0a-8304-c28c20f54d10)

**EDIT:** That's with a ZIP theme placed in the following path: `\Android\data\gr.talent.cruiser\files\maps\_themes` If the ZIP has multiple XML "theme" files, you get additional entries to pick from.

**Reply by devemux86** [reply]
@MotoUKRider very nice hint! :+1: 

The "themes" folder is monitored because of the downloaded map themes.

If we place a theme zip file there, it will also appear in the "Map theme" selection screen.

**Reply by DIrgi** [reply]
Well, how should one know?

**Reply by DIrgi** [reply]
Sorry but where the heck is the theme folder on my android? 
Can you deliver more input i.e. a more complete path?

Or do I have to create a folder "themes" first and place the zip file there? - What I did btw. but nothing appeared or happened
see screen shot 

![image](https://github.com/devemux86/cruiser/assets/57968526/4857ec5e-ce15-4e67-b774-784107c98f1d)

**Reply by 0709wiwiwi** [reply]
![themes](https://github.com/devemux86/cruiser/assets/54167201/905f9e88-64c2-406f-ad1a-21f00d8d3e38)

**Reply by devemux86** [reply]
@DIrgi download a theme with the download manager and the folder structure will be created automatically.

The "themes" folder is inside the main folder, together with "maps", etc.

**Reply by DIrgi** [reply]
Thanks, now we're getting closer to the matter...
I have an empty folder so far...see shot:

![image](https://github.com/devemux86/cruiser/assets/57968526/2d0e70a7-b2dd-4996-93c5-7b7f954910f6)

My main data is there:
under Android/data
and not Android/media (like you have)
and even under Android/data, theres the maps folder but within no themes, just the mapsforge folder with the mapsforge stuff

![image](https://github.com/devemux86/cruiser/assets/57968526/4fb76f4e-ace8-405d-97e5-3d68508fd748)

**Reply by 0709wiwiwi** [reply]
Download a single open andromap and the folder will be created. 
(Your Cruiser app advanced setting)  
- Find the themes folder Under Android | data | gr.talent.cruiser | files 

- Or if set as such in the Cruiser app under Android | media  | gr.talent.cruiser |  (Allows file acces in A13+ !)

![image](https://github.com/devemux86/cruiser/assets/54167201/f41328bb-47d6-413d-a1c3-23137c3be9f4)

**Reply by WolfgangZ42** [reply]
Try "_themes"
![image](https://github.com/devemux86/cruiser/assets/121806238/43b3cadf-2eae-4edf-afc1-7e48d7326744)

**Reply by 0709wiwiwi** [reply]
@WolfgangZ42 
With the different versions, the paths where the downloads end up have changed at times. 
That's also why it is so damned difficult if not impossible to produce a wiki manual right now ;-)

**Reply by DIrgi** [reply]
Strange, no way.....

first of all  Android/media is not in us at all by the cruiser App.
So on my system everything is beyond Android/data
anyway....

following *wiwi and wolfgang, I downloaded some Openandro stuff and yes, after Cruiser restart
a themes folder is now on the same level like the maps folder:
<img width="539" alt="image" src="https://github.com/devemux86/cruiser/assets/57968526/a10f4c1f-fd89-4ee9-ad32-3c49dfd43b0d">

but not like Wolfgang (within) the maps folder

![image](https://github.com/devemux86/cruiser/assets/57968526/9196adb2-a59d-4ab2-89de-727444873344)
![image](https://github.com/devemux86/cruiser/assets/57968526/c5f50736-2f15-42db-b465-8b61ccf3e8b0)

Now I see "Elevate" stuff for selection in the themes options:
<img width="413" alt="image" src="https://github.com/devemux86/cruiser/assets/57968526/48bd4979-3551-45a3-9d73-337ef2e486a7">


Ok works somehow but......

what I did next was to create a test folder and saved my file(s) there.
but.....Cruiser do not look for them or load them. Works only  in the way when I do it the manual way, I did before (see thread start).

I deleted all stuff out of andromaps (because I don't need it)
What I recognized then: When I put my files now in the empty andromaps folder they will be found and used by Cruiser.
I assume Cruiser does internally a registry of files, downloaded via the Cruiser menu but NOT look for other folders)

Wrap Up:
And coming back to my initial request.

Although it works somehow (via the backdoor) this should not be the case for a common user and why I should have a "andromaps dead file folder" just to put my files there that that it works...? - And if I never plan to have andromaps (or freizeitkarte or what ever), Cruiser will never look into my own mnual created folder there for some themes.

Best should be load an own theme file (as initially suggested) via the cruiser surface and cruiser does the rest like copying into the right folder and connect it with the options.

**Reply by devemux86** [reply]
> Best should be load an own theme file (as initially suggested) via the cruiser surface and cruiser does the rest like copying into the right folder and connect it with the options.

This seems a simpler solution, as everything should be done with app interface.

(the other one is a workaround, if users like to do things with the file system)

**Reply by 0709wiwiwi** [reply]
@DIrgi 
 I sure agree that it could be offered more simply of course.
It is also true that themes are developed for certain maps.
FZOAM developed for openandromaps does not opimally fit Mapsforge.
Considering was added through the back door, I did not report this either.

**Reply by 0709wiwiwi** [reply]
@DIrgi 

Advantage Cruiser set up into Android | media on A13+.
Android | media keeps accessible trought the file explorer.
Allows sharing of the vector maps with Locus & Greenalp RTT2.

Precaution should you ever deinstall Cruiser for any reason.
Any reason. As damaging Cruiser not able to start up.
In any case, I've managed to do that once ;-)

Change Android (media) path (add old) than deinstall.
After new install restore your data. (GB's sized maps)

The next proposal did not work :-( Do NOT use keep data !
https://github.com/devemux86/cruiser/discussions/80
The intention was good, but the result was a disaster.

**Reply by devemux86** [reply]
#80 is just a flag, the functionality is provided by Android system, apps cannot override it.

The media folder remains accessible by file managers, until Google decides otherwise in new Androids. :slightly_smiling_face:

**Reply by 0709wiwiwi** [reply]
>Until Google decides otherwise.  🙂

Google the new emperor get a 👍 or 👎
